### PR TITLE
[notebook] scale-test always cleans up

### PR DESCRIPTION
**Pull request by danking**
cc: @tpoterba 

No matter what happens, always grab a CSRF from the notebook leader and ask to delete my notebook.

### CODE REVIEW

**Comment by danking** [comment]
This also means you don't have to delete every notebook that's running (i.e. `k delete -l app=notebook-worker`).

File: notebook/scale-test.py [hunk 1]
```diff
@@ -26,86 +26,88 @@ def get_cookie(session, name):
...
-        data = aiohttp.FormData()
-        data.add_field(name='_csrf', value=get_cookie(session, '_csrf'))
-        async with session.post(
-                deploy_config.url('workshop', '/notebook/delete'),
-                data=data,
-                headers=headers) as resp:
-            pass
-
-        log.info(f'{i} notebook delete, done.')
+        try:
```
Comment (cseed) [inline notebook/scale-test.py]: You need to set duration = None, ready = False before the try or you'll get an undefined variable error in the case of exception.

Comment (danking) [reply notebook/scale-test.py]: I don't suppress the exception, I just make sure we delete the notebook.

Do you want to catch the exception and return None, None?

Comment (cseed) [reply notebook/scale-test.py]: That would also work.  You return after the try ... except block and set those variables that are set only in the body of the try ... except.

Comment (cseed) [reply notebook/scale-test.py]: Oh, oops, I see what you are saying.

Yes, we should supress the exception and return None, None.  A not true ready is treated as an error by main.  Either this, or it needs to be modified to handle the exceptions generated in teh children of the gather.

Comment (cseed) [reply notebook/scale-test.py]: To clarify further, this fixes the cleanup, but it doesn't treat exceptions in the made code as failed notebook attempts.  Can you make this change also?

**Comment by cseed** [comment]
I'm going to take this over.

**Pull request closed without merging**